{{cookiecutter.project_name}}/.azure/devops-pipelines/terraform-cicd.yml

```diff
@@ -0,0 +1,399 @@
+      displayName: 'Terraform Plan'
+      workingDirectory: $(working-directory)/$(environment)
+      enabled: 'true'
+      env:
+        ARM_TENANT_ID: $(STAGING_AZURE_SP_TENANT_ID)
+        ARM_CLIENT_ID: $(STAGING_AZURE_SP_APPLICATION_ID)
+        ARM_CLIENT_SECRET: $(STAGING_AZURE_SP_CLIENT_SECRET)
+        DATABRICKS_HOST: $(STAGING_DATABRICKS_HOST)
+
+  - job: ProdTerraformCI
+    displayName: 'Continuous integration tests for prod Terraform scripts'
+    dependsOn: []   # Removes the implicit dependency on previous job and force ProdTerraformCI job to run in parallel
+
+    pool:
+      vmImage: 'ubuntu-latest'
+
+    variables:
+      environment: prod
+
+    steps:
+    - script: env | sort
+      displayName: 'Environment / Context'
+
+    - checkout: self
+      displayName: 'Checkout & Build.Reason: $(Build.Reason) & Build.SourceBranchName: $(Build.SourceBranchName)'
+      persistCredentials: true
+      clean: true
+
+    - script: |
+        set -e
+        PROD_AZURE_SP_TENANT_ID=$(PROD_AZURE_SP_TENANT_ID)
+        PROD_AZURE_SP_APPLICATION_ID=$(PROD_AZURE_SP_APPLICATION_ID)
+        PROD_AZURE_SP_CLIENT_SECRET=$(PROD_AZURE_SP_CLIENT_SECRET)
+        DATABRICKS_TOKEN=$(.azure/devops-pipelines/scripts/generate-aad-token.sh "$PROD_AZURE_SP_TENANT_ID" "$PROD_AZURE_SP_APPLICATION_ID" "$PROD_AZURE_SP_CLIENT_SECRET")
+        echo "##vso[task.setvariable variable=DATABRICKS_TOKEN;issecret=true]${DATABRICKS_TOKEN}"
+      displayName: 'Configure AAD auth'
+
+    - script: |
+        eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"
+        brew tap hashicorp/tap
+        brew install hashicorp/tap/terraform
+      displayName: 'Install Terraform'
+
+    - script: |
+       terraform fmt -check -recursive -diff
+      continueOnError: true
+      displayName: 'Terraform fmt'
+
+    - script: |
+        terraform init
+      displayName: 'Terraform Init'
+      workingDirectory: $(working-directory)/$(environment)
+      enabled: 'true'
+      env:
+        ARM_TENANT_ID: $(PROD_AZURE_SP_TENANT_ID)
+        ARM_CLIENT_ID: $(PROD_AZURE_SP_APPLICATION_ID)
+        ARM_CLIENT_SECRET: $(PROD_AZURE_SP_CLIENT_SECRET)
+
+    - script: |
+        terraform validate -no-color
+      displayName: 'Terraform Validate'
+      workingDirectory: $(working-directory)/$(environment)
+      enabled: 'true'
+
+    - script: |
+        terraform plan -no-color \
+        -var="git_repo_url=$(Build.Repository.Uri)"
+      displayName: 'Terraform Plan'
+      workingDirectory: $(working-directory)/$(environment)
+      enabled: 'true'
+      env:
+        ARM_TENANT_ID: $(PROD_AZURE_SP_TENANT_ID)
+        ARM_CLIENT_ID: $(PROD_AZURE_SP_APPLICATION_ID)
+        ARM_CLIENT_SECRET: $(PROD_AZURE_SP_CLIENT_SECRET)
+        DATABRICKS_HOST: $(PROD_DATABRICKS_HOST)
+
+# Run Terraform CD stage after successfully merging to {{cookiecutter.default_branch}}
+- stage: TerraformCD
+  displayName: 'Terraform Deployment for mlops-stack-ado'
```

Comment:
The pipeline name here probably should include the `{{cookiecutter.project_name}}` rather than `mlops-stack-ado`?

Reply:
Same comment for other instances of `mlops-stack-ado` in this PR (maybe we can find-and-replace with `{{cookiecutter.project_name}}`)

Reply:
good catch, thought I had indeed done a find and replace on those!